pubtools/_quay/operator_pusher.py

```diff
@@ -453,25 +453,73 @@ def build_index_images(self):
+            non_fbc_items = []
+            osev_tuple = tuple([int(x) for x in version.replace("v", "").split(".")])
+            for item in items:
+                for repo in item.metadata["tags"]:
+                    if repo not in repos_opted_in:
+                        repos_opted_in[repo] = pyxis_get_repo_metadata(
+                            repo, self.target_settings
+                        ).get("fbc_opt_in", False)
+                item_fbc_opt_in = [repos_opted_in[repo] for repo in item.metadata["tags"]]
+                if not (all(item_fbc_opt_in) or not any(item_fbc_opt_in)):
+                    failed_items[id(item)] = True
+                    item.add_error("NOTPUSHED", "Inconsistent fbc opt in")
+                    LOG.error("Inconsistent fbc opt in for item {i}".format(i=item))
+                    continue
+                elif all(item_fbc_opt_in):
+                    items_opted_in[id(item)] = True
+                else:
+                    items_opted_in[id(item)] = False
+
+                if not items_opted_in[id(item)] and osev_tuple >= (4, 11):
+                    LOG.error(
+                        "{i} is not opted in FBC but target OSE is {ose}".format(
+                            i=item, ose=version
+                        )
+                    )
+                    item.add_error(
+                        "UPLOADFAILED",
+                        "not opted in FBC but target OSE is {ose}".format(ose=version),
+                    )
+                    failed_items[id(item)] = True
+                elif not items_opted_in[id(item)] or (
+                    items_opted_in[id(item)] and osev_tuple <= (4, 10)
+                ):
+                    non_fbc_items.append(item)
+                elif items_opted_in[id(item)] and osev_tuple >= (4, 11):
+                    LOG.warning(
+                        "Skipping item {i} from iib build {ose} as it's opted in for FBC".format(
+                            i=item, ose=version
+                        )
+                    )
+
+            is_hotfix = any([item.metadata.get("com.redhat.hotfix") for item in non_fbc_items])
             is_advisory_source = all(
-                [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in items]
+                [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in non_fbc_items]
             )
             item_groups = {}
             if is_hotfix and not is_advisory_source:
                 raise ValueError("Cannot push hotfixes without an advisory")
             if is_hotfix:
-                for item in items:
+                for item in non_fbc_items:
                     item_groups.setdefault(item.origin, []).append(item)
             else:
-                item_groups["default"] = items
+                item_groups["default"] = [
+                    item for item in non_fbc_items if not id(item) in failed_items
```

Comment:
can an item be in non_fbc_items and in failed_items at the same time?

Reply:
yeah, that condition here is redundant